sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -3275,16 +3312,41 @@ pi_result piDevicePartition(pi_device Device,
     return Res;
   }
 
+  auto EffectiveSize = [&]() -> decltype(Device->SubDevices.size()) {
```

Comment:
why do you need a lambda function here?

Reply:
Just an idiom (Immediately Invoked Function Expression) for complex variable initialization. Also allows to use "early return" idiom to increase readability.

Reply:
OK, but name the variable more clear